sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemPropertiesTable.java

```diff
@@ -0,0 +1,174 @@
+      ObjectMapper jsonMapper
+  )
+  {
+    this.druidNodeDiscoveryProvider = druidNodeDiscoveryProvider;
+    this.authorizerMapper = authorizerMapper;
+    this.httpClient = httpClient;
+    this.jsonMapper = jsonMapper;
+  }
+
+  @Override
+  public RelDataType getRowType(RelDataTypeFactory typeFactory)
+  {
+    return RowSignatures.toRelDataType(PROPERTIES_SIGNATURE, typeFactory);
+  }
+
+  @Override
+  public Schema.TableType getJdbcTableType()
+  {
+    return Schema.TableType.SYSTEM_TABLE;
+  }
+
+  @Override
+  public Enumerable<Object[]> scan(DataContext root)
+  {
+    final AuthenticationResult authenticationResult = (AuthenticationResult) Preconditions.checkNotNull(
+        root.get(PlannerContext.DATA_CTX_AUTHENTICATION_RESULT),
+        "authenticationResult in dataContext"
+    );
+    SystemSchema.checkStateReadAccessForServers(authenticationResult, authorizerMapper);
+    final Iterator<DiscoveryDruidNode> druidServers = SystemSchema.getDruidServers(druidNodeDiscoveryProvider);
+
+    final Map<String, Pair<StringBuilder, Stream<Object[]>>> serverToPropertiesMap = new HashMap<>();
+    druidServers.forEachRemaining(discoveryDruidNode -> {
+      final DruidNode druidNode = discoveryDruidNode.getDruidNode();
+      final Map<String, String> propertiesMap = getProperties(druidNode);
+      if (serverToPropertiesMap.containsKey(druidNode.getHostAndPortToUse())) {
+        Pair<StringBuilder, Stream<Object[]>> pair = serverToPropertiesMap.get(druidNode.getHostAndPortToUse());
+        pair.lhs.append(StringUtils.format(",%s", discoveryDruidNode.getNodeRole().getJsonName()));
+      } else {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(StringUtils.format("%s", discoveryDruidNode.getNodeRole().getJsonName()));
+        serverToPropertiesMap.put(druidNode.getHostAndPortToUse(), Pair.of(builder, propertiesMap.entrySet().stream()
+                                                                                                              .map(entry -> new Object[]{
+                                                                                                                    druidNode.getServiceName(),
+                                                                                                                    druidNode.getHostAndPortToUse(),
+                                                                                                                    entry.getKey(),
+                                                                                                                    entry.getValue()
+                                                                                                              })
+            )
+        );
+      }
+    });
+    return Linq4j.asEnumerable(serverToPropertiesMap.values().stream().flatMap(pair -> pair.rhs.map(entry -> new Object[]{entry[0], entry[1], pair.lhs.toString(), entry[2], entry[3]})).collect(Collectors.toList()));
```

Comment:
Please check if this can be formatted better by breaking up the argument of `asEnumerable` into multiple lines.